roles/openshift_hosted/tasks/registry/registry.yml

```diff
@@ -0,0 +1,40 @@
+---
+- name: Retrieve list of openshift nodes matching registry selector
+  command: >
+    {{ openshift.common.client_binary }} --api-version='v1' -o json
+    get nodes -n default --config={{ openshift_hosted_kubeconfig }}
+    --selector={{ openshift.hosted.registry.selector | default('') }}
+  register: openshift_hosted_registry_nodes_json
+  changed_when: false
+  when: openshift.hosted.registry.replicas | default(none) is none
```

Comment:
Let me know if that's infeasible or just outright a terrible idea, I'd just love to see us get away from some situations where the same little bit of data is desired state, and we refer to it by 3-4 different variables/locations throughout.